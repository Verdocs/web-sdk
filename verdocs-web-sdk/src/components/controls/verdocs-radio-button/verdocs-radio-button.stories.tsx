import {html} from 'lit-html';
import {Meta} from '@storybook/web-components';

export default {
  title: 'Controls/Radio Button',
  component: 'verdocs-radio-button',
  args: {
    checked: false,
    name: 'test',
    value: '1',
    disabled: false,
  },
  argTypes: {
    input: {
      action: 'input',
      description:
        'Fired when the radio button is clicked. e.target.checked will always be true, because radio button onChange events are fired only when the radio button is clicked to select it.',
    },
  },
} as Meta;

export const RadioButton = ({checked, name, value, disabled, input}) =>
  html` <verdocs-radio-button .checked=${checked} .name=${name} .value=${value} .disabled="${disabled}" @input=${input} /> `;
