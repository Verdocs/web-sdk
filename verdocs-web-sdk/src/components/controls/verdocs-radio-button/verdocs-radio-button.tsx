import {Component, h, Host, Prop} from '@stencil/core';

/**
 * Displays a radio button. Note that this is different from the `verdocs-field-radio-button` component, which is
 * designed to be used in signing experiences and contains settings that connect to template fields. This is just a
 * simple radio button for UI displays e.g. dialog boxes.
 *
 * This control encapsulates a standard HTML radio button. To subscribe to change events, connect an `onChange`
 * handler. Sample usage:
 *
 * ```ts
 * <verdocs-radio-button
 *    value="val1"
 *    name="someProperty"
 *    checked={this.someProperty === 'val1'}
 *    onInput={(e: any) => { this.someProperty = 'val1' }}
 *    disabled={false}
 * />
 * ```
 */
@Component({
  tag: 'verdocs-radio-button',
  styleUrl: 'verdocs-radio-button.scss',
  shadow: false,
})
export class VerdocsRadioButton {
  /**
   * Whether the radio button is currently selected.
   */
  @Prop() checked: boolean = false;

  /**
   * HTML form field name for the input.
   */
  @Prop() name: string = '';

  /**
   * Value to track with the input.
   */
  @Prop() value: string = '';

  /**
   * If set, the button will still be displayed but not selectable.
   */
  @Prop() disabled?: boolean = false;

  render() {
    return (
      <Host class={{disabled: this.disabled}}>
        <input
          type="radio"
          value={this.value}
          name={this.name}
          id={`verdocs-radio-button-${this.name}-${this.value}`}
          checked={this.checked}
          disabled={this.disabled}
          // onChange={e => this.handleChange(e)}
        />
        <label htmlFor={`verdocs-radio-button-${this.name}-${this.value}`} />
      </Host>
    );
  }
}
