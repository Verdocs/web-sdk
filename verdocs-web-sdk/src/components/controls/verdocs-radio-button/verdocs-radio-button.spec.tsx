import { newSpecPage } from '@stencil/core/testing';
import { VerdocsRadioButton } from './verdocs-radio-button';

describe('verdocs-radio-button', () => {
  it('renders with default props', async () => {
    const page = await newSpecPage({
      components: [VerdocsRadioButton],
      html: `<verdocs-radio-button></verdocs-radio-button>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('renders as checked and disabled', async () => {
    const page = await newSpecPage({
      components: [VerdocsRadioButton],
      html: `<verdocs-radio-button checked="true" disabled="true" value="val1" name="group1"></verdocs-radio-button>`,
    });
    expect(page.root).toMatchSnapshot();
    const input = page.root.querySelector('input[type="radio"]') as HTMLInputElement | null;
    expect(input?.checked).toBe(true);
    expect(input?.disabled).toBe(true);
  });

  it('renders with value and name', async () => {
    const page = await newSpecPage({
      components: [VerdocsRadioButton],
      html: `<verdocs-radio-button value="val2" name="group2"></verdocs-radio-button>`,
    });
    expect(page.root).toMatchSnapshot();
    const input = page.root.querySelector('input[type="radio"]') as HTMLInputElement | null;
    expect(input?.value).toBe('val2');
    expect(input?.name).toBe('group2');
  });
});
