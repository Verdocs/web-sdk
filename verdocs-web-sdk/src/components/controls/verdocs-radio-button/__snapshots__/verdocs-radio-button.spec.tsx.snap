// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`verdocs-radio-button renders as checked and disabled 1`] = `
<verdocs-radio-button checked="true" class="disabled" disabled="true" name="group1" value="val1">
  <input checked="" disabled="" id="verdocs-radio-button-group1-val1" name="group1" type="radio" value="val1">
  <label htmlfor="verdocs-radio-button-group1-val1"></label>
</verdocs-radio-button>
`;

exports[`verdocs-radio-button renders with default props 1`] = `
<verdocs-radio-button>
  <input id="verdocs-radio-button--" type="radio" value="">
  <label htmlfor="verdocs-radio-button--"></label>
</verdocs-radio-button>
`;

exports[`verdocs-radio-button renders with value and name 1`] = `
<verdocs-radio-button name="group2" value="val2">
  <input id="verdocs-radio-button-group2-val2" name="group2" type="radio" value="val2">
  <label htmlfor="verdocs-radio-button-group2-val2"></label>
</verdocs-radio-button>
`;
